src/erl_cache_decorator.erl

```diff
@@ -10,10 +10,10 @@
     (fun(() -> term())).
 cache_pt(Fun, Args, {Module, FunctionAtom, Name, Opts}) ->
     FinalOpts = [{refresh_callback, fun () -> Fun(Args) end} | Opts],
-    Key = case proplists:get_value(key_generation, Opts) of
-        Module when is_atom(Module) ->
-            apply(Module, generate_key, [Name, Module, FunctionAtom, Args]);
-        undefined ->
+    Key = case proplists:lookup(key_generation, Opts) of
+        {_, KeyModule} when is_atom(KeyModule) ->
```

Comment:
should this also have the guard `, KeyModule /= undefined`?

Reply:
It's the name of my module!

Added.